cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/leveldb/LevelDBStreamConsumerStateStoreFactory.java

```diff
@@ -30,26 +39,38 @@
  * Factory for creating {@link StreamConsumerStateStore} in level db.
  */
 public final class LevelDBStreamConsumerStateStoreFactory implements StreamConsumerStateStoreFactory {
-
   private final LevelDBTableService tableService;
+  private final CConfiguration cConf;
 
   @Inject
-  LevelDBStreamConsumerStateStoreFactory(LevelDBTableService tableService) {
+  LevelDBStreamConsumerStateStoreFactory(LevelDBTableService tableService, CConfiguration cConf) {
     this.tableService = tableService;
+    this.cConf = cConf;
   }
 
   @Override
   public synchronized StreamConsumerStateStore create(StreamConfig streamConfig) throws IOException {
     Id.Namespace namespace = streamConfig.getStreamId().getNamespace();
-    String tableName = StreamUtils.getStateStoreTableName(namespace);
-    tableService.ensureTableExists(tableName);
-    LevelDBTableCore coreTable = new LevelDBTableCore(tableName, tableService);
+    TableId tableId = StreamUtils.getStateStoreTableId(namespace);
+
+    getLevelDBTableAdmin(tableId).create();
+    String levelDBTableName =
+      PrefixedNamespaces.namespace(cConf, tableId.getNamespace().getId(), tableId.getTableName());
+    LevelDBTableCore coreTable = new LevelDBTableCore(levelDBTableName, tableService);
     return new LevelDBStreamConsumerStateStore(streamConfig, coreTable);
```

Comment:
I feel that ``LevelDBStreamConsumerStateStore`` should take a ``LevelDBTable`` and not a ``LevelDBCoreTable``. That way you don't need to create the namespaced name here. ``InMemoryStreamConsumerStateStore`` and ``HBaseStreamConsumerStateStore`` follow a similar pattern.